server/src/termdb.gdc.js

```diff
@@ -998,8 +999,20 @@ async function cacheSampleIdMapping(ds) {
 	initGdcHolder(ds)
 
 	try {
+		// also indicates to ignore data_release_version even if present
+		const response = await ds.preInit.getStatus()
+		if (deepEqual(response.data_release_version, ds.__gdc.data_release_version) && ds.__gdc.doneCaching) {
+			// do not trigger another caching if it's been done for saved release version
```

Comment:
i think this implementation simplifies things. just rename this func to cacheMappingOnNewRelease()

Reply:
I ended up doing a more thorough cleanup of the caching check and code, mostly to ensure race condition does not cause a stale cache to be used.